ui/src/main/webapp/admin-tools/wcpm/index.js

```diff
@@ -0,0 +1,254 @@
+import React from 'react'
+import { connect } from 'react-redux'
+import { compose, gql, graphql } from 'react-apollo'
+
+import { fromJS } from 'immutable'
+
+import Snackbar from 'material-ui/Snackbar'
+
+import {
+  Panel,
+  H1,
+  Disabled
+} from './components'
+
+import Description from 'components/Description'
+
+import {
+  isEditing,
+  edit,
+  cancel,
+  update,
+  submit,
+  error,
+  dismiss,
+  getValue,
+  editingToken,
+  hasSubmitted,
+  getNotification,
+  getErrors
+} from './reducer'
+
+const NotifyErrors = connect(
+  (state) => ({ message: getNotification(state) }),
+  { onClose: dismiss }
+)(
+  ({ message, onClose }) =>
+    <Snackbar
+      open={message !== null}
+      message={message || ''}
+      autoHideDuration={5000}
+      onRequestClose={onClose} />
+)
+
+import Whitelist, { WhitelistEdit, validator as whitelistValidator } from './whitelist'
+
+import policyValidator from './validator'
+import Policy, { PolicyEdit, NewPolicy } from './policy'
+
+const connector = (key, validator) => {
+  return connect(
+    (state, { policies, whitelisted }) => {
+      const value = getValue(state)
+      return {
+        [key]: value,
+        serverErrors: getErrors(state),
+        errors: hasSubmitted(state)
+          ? validator(value, { policies, whitelisted }).toJS() : undefined
+      }
+    },
+    { onUpdate: update, onCancel: cancel }
+  )
+}
+
+const ConnectedWhitelistEdit = connector('whitelisted', whitelistValidator)(WhitelistEdit)
+const ConnectedPolicyEdit = connector('policy', policyValidator)(PolicyEdit)
+
+const insert = (value, index, array) => array.slice(0, index).concat(value, array.slice(index + 1))
+
+let WebContextPolicyManager = (props) => {
+  const {
+    data,
+
+    whitelisted = [],
+    policies = [],
+
+    editing,
+    index,
+
+    onEdit,
+    onDelete,
+    onSavePolicy,
+    onSaveWhitelist
+  } = props
+
+  const mergedPolicies = (editing && index > -1) ? insert({ editing: true }, index, policies) : policies
+
+  const ps = mergedPolicies.map((policy, i) => {
+    if (editing && policy.editing) {
+      return (
+        <ConnectedPolicyEdit
+          data={data}
+          number={i + 1}
+          policies={mergedPolicies}
+          whitelisted={whitelisted}
+          onSave={(policy) => onSavePolicy(policy, i)}
+          onDelete={() => onDelete(i)} />
+      )
+    } else {
+      return (
+        <Policy
+          number={i + 1}
+          policy={policy}
+          onEdit={(value) => onEdit(i, value)} />
+      )
+    }
+  })
+
+  const applyDisabled = (el, i) =>
+    (!editing || index === (i - 1)) ? el : <Disabled>{el}</Disabled>
+  const applyPanel = (el, i) => <Panel>{el}</Panel>
+  const applyKey = (el, i) => <div key={i}>{el}</div>
+
+  const els = [
```

Comment:
A list of elements for lack of a better name. It's a short lived variable so I don't think it needs a great name, but if you have a short and concise name, I would be okay changing it.